oat-sa/tao-item-runner-fe
# Feature/ADF-1417/Upgrade to Node 18

### Related to: [ADF-1417](https://oat-sa.atlassian.net/browse/ADF-1417) and [ADF-1419](https://oat-sa.atlassian.net/browse/ADF-1419)

Requires: 
 - [x] update the file [`package.json`](https://github.com/oat-sa/tao-item-runner-fe/blob/37fcf505332409243261c2f05832e473950ce9c5/package.json#L46-L49) once the companion PRs have been merged and released

### Summary

Update the devDependencies for supporting Node 18

Set Node 18 for the CI workflow.

### Details

To properly support Node 18, the dev dependencies had to be updated and aligned.

**Note:** The lock file was generated with `npm@6` so it would also work seamlessly with Node 14.

### How to test
- check the report on GitHub
- make sure to have Node 18 installed (you can use [nvm.sh](https://github.com/nvm-sh/nvm) for managing multiple versions)
- install the package: `npm i`
- run the unit tests: `npm test`, `npm run test:cov`, `npm run coverage`
- test the bundling: `npm run build`
- test the linting: `npm run lint`


[ADF-1417]: https://oat-sa.atlassian.net/browse/ADF-1417?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
[ADF-1419]: https://oat-sa.atlassian.net/browse/ADF-1419?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
